src/api/customers.ts

```diff
@@ -0,0 +1,94 @@
+import type { ApiRoot, ClientResponse, Customer, CustomerDraft, CustomerToken } from '@commercetools/platform-sdk';
+import { createApiBuilderFromCtpClient } from '@commercetools/platform-sdk';
+import { ctpClient } from '@/api/commerceToolsClient';
+import { AuthData as AUTH } from '@/api/token/authData';
+import type { SignInResponse } from '@/data/interfaces';
+
+const apiRoot: ApiRoot = createApiBuilderFromCtpClient(ctpClient);
+
+export async function registerCustomer(customerDraft: CustomerDraft): Promise<SignInResponse> {
+  try {
+    const response: ClientResponse<SignInResponse> = await apiRoot
+      .withProjectKey({ projectKey: AUTH.projectKey })
+      .customers()
+      .post({ body: customerDraft })
+      .execute();
+    return response.body;
+  } catch (error) {
+    console.error('Failed to register user:', error);
+    throw error;
+  }
+}
+
+export const loginCustomer: (email: string, password: string) => Promise<SignInResponse> = async (
+  email: string,
+  password: string
+): Promise<SignInResponse> => {
```

Comment:
```suggestion
export const loginCustomer = async (
  email: string,
  password: string
): Promise<SignInResponse> => {
```
it would be enough, since we are already defining types there